### ISSUE TITLE
Roslyn should assume the worst nullability when a variable mentioned in multiple byref arguments

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#36745]
**Version Used**: 

```
Branch master (21 Jun 2019)
Latest commit 898bed by Heejae Chang:
added NFW to get some data on incremental parsing bug where source si� (#36620)

* added NFW to get some data on incremental parsing bug where source size and tree size is different

* more comments
```

**Steps to Reproduce**:

Compile and run the following code:
```cs
#nullable enable
using System;
public class C {
    
    static void Main()
    {
        new C().M("");
    }
    
    void M(string? s)
    {
        M(out s, out s);
        s.ToString();
    }

    void M(out string? s1, out string s2)
    {
        s2 = "";
        s1 = null;
    }
}
```

**Expected Behavior**:
Warning for `s.ToString()` since `s` is passed to an out parameter which is explicitly declared as nullable.

**Actual Behavior**:
Roslyn does not report any warnings for the code above. The program crashes at runtime with a `NullReferenceException`

**Notes**
Roslyn should merge nullability annotations for all parameters effectively using the worst of available annotations since there's no way to know in which order out parameters will be assigned. Currently Roslyn uses the last annotation i.e. the warning will appear if you reverse the order of parameters in `M` since the nullable parameter will become the last.

There's a similar problem with `ref` parameters. If you change `out` to `ref` in the code above whether you'll receive a warning for `s.ToString()` or not depends on the order of `ref` parameters in `M`. Although there's always an additional warning for passing `string?` to `ref string` in this case, I believe that the warning for `s.ToString()` should not depend on the declaration of `M`

**Comment by jcouv** [comment]
Relates to https://github.com/dotnet/roslyn/issues/33587 (possible duplicate)

**Comment by TessenR** [comment]
Note that similar problem exits with flow annotations. The following code does not any warnings but crashes at runtime with a null reference exception. I believe Roslyn should be able to detect that `s` is also passed to a nullable parameter which doesn't have any contract annotations.

```cs
#nullable enable
using System.Diagnostics.CodeAnalysis;
public class C {
    public static bool M(out string? s1, [NotNullWhen(true)] out string? s2) {
        s2 = "";
        s1 = null;
        return true;
    }
    
    static void Main()
    {
        string? s;
        if (M(out s, out s))
            s.ToString();
    }
}
```

**Comment by jaredpar** [comment]
Closing as a dupe of #33587. Will make sure to validate the extra scenario here around flow annotations when resolving.
